znai-java/src/main/java/com/twosigma/znai/java/extensions/JavaIncludePlugin.java

```diff
@@ -68,35 +74,60 @@ public JavaIncludeResult process(JavaCode javaCode) {
         return new JavaIncludeResult(Collections.singletonList(docElement), snippet);
     }
 
-    private String extractContent(JavaCode javaCode, Boolean isBodyOnly, Boolean isSignatureOnly) {
+    private String extractContent(JavaCode javaCode) {
         if (entry == null && entries == null) {
             return javaCode.getFileContent();
         }
 
         Stream<String> methodNamesWithOptionalTypes = entry != null ? Stream.of(entry) : entries.stream();
-        return methodNamesWithOptionalTypes.map(n -> extractSingleContent(javaCode, n, isBodyOnly, isSignatureOnly))
-                .collect(Collectors.joining(isSignatureOnly ? "\n" : "\n\n"));
+        return methodNamesWithOptionalTypes
+                .map(nameWithOptionalType -> extractSingleEntryContent(javaCode, nameWithOptionalType))
+                .collect(collectorWithSeparator());
     }
 
-    private String extractSingleContent(JavaCode javaCode, String entry, Boolean isBodyOnly, Boolean isSignatureOnly) {
+    private String extractSingleEntryContent(JavaCode javaCode, String entry) {
         return javaCode.hasType(entry) ?
-                extractTypeContent(javaCode, entry, isBodyOnly) :
-                extractMethodContent(javaCode, entry, isBodyOnly, isSignatureOnly);
+                extractTypeContent(javaCode, entry) :
+                extractMethodContent(javaCode, entry);
     }
 
-    private String extractTypeContent(JavaCode javaCode, String entry, Boolean isBodyOnly) {
+    private String extractTypeContent(JavaCode javaCode, String entry) {
         JavaType type = javaCode.findType(entry);
         return isBodyOnly ? type.getBodyOnly() : type.getFullBody();
     }
 
-    private String extractMethodContent(JavaCode javaCode, String entry, Boolean isBodyOnly, Boolean isSignatureOnly) {
-        JavaMethod method = javaCode.findMethod(entry);
+    private String extractMethodContent(JavaCode javaCode, String entry) {
+        List<JavaMethod> methods = isMultipleEntries ?
+                javaCode.findAllMethods(entry) :
+                Collections.singletonList(javaCode.findMethod(entry));
 
-        return isBodyOnly ?
-                extractBodyOnly(method) :
-                isSignatureOnly ? method.getSignatureOnly() :
-                        method.getFullBody();
+        if (isBodyOnly) {
```

Comment:
Perhaps there should be a warning printed out to the log if isBodyOnly && isMultipleEntries, since that would probably result in confusing documentation.

Reply:
Or if the separator for that case can make it clear that the bodies come from different methods (see my comment below), then this warning is not necessary.